microcosm/typing.py

```diff
@@ -0,0 +1,8 @@
+from typing import Callable, Mapping
+
+from microcosm.config.model import Configuration
+from microcosm.metadata import Metadata
+
+
+Loader = Callable[[Metadata], Mapping]
+SecondaryLoader = Callable[[Metadata, Configuration], Mapping]
```

Comment:
I thought pretty carefully about these definitions.  Due to covariance / contravariance, I decided that the return value types should be the most general possible, while the parameter types should be the most specific.  This setup results in these types being as permissive as possible.  Then within `two_stage_loader`, we wrap the outputs of these loaders in `Configuration` so that `two_stage_loader` returns the most specific type possible, so that it can be used in as many situations as possible.